server/src/main/java/io/crate/auth/HttpAuthUpstreamHandler.java

```diff
@@ -112,7 +112,7 @@ private void handleHttpRequest(ChannelHandlerContext ctx, HttpRequest request) {
             sendUnauthorized(ctx.channel(), errorMessage);
         } else {
             try {
-                Role user = authMethod.authenticate(username, credentials.password(), connectionProperties);
+                Role user = authMethod.authenticate(credentials, connectionProperties);
```

Comment:
`credentials.setUsername` will be done before passing on credentials => when JWT flow is implemented